src/ngx_http_lua_subrequest.c

```diff
@@ -382,15 +404,62 @@ ngx_http_lua_ngx_location_capture_multi(lua_State *L)
 
             type = lua_type(L, -1);
 
-            if (type == LUA_TNIL) {
-                method = NGX_HTTP_GET;
+            switch (type) {
+                case LUA_TNIL:
+                    method = NGX_HTTP_GET;
+                    break;
+                case LUA_TNUMBER:
+                    method = (ngx_uint_t) lua_tonumber(L, -1);
+                    break;
+                case LUA_TSTRING:
+                    lua_method_name = (u_char *) lua_tolstring(L, -1,
+                                                        &method_name_length);
+                    if (method_name_length == 0) {
+                        return luaL_error(L, "Bad http request method");
+                    }
+                    
+                    methods_number = sizeof(ngx_http_lua_ordered_methods) /
+                        sizeof(ngx_http_lua_ordered_methods[0]);
+                    for (method_index = 0; method_index < methods_number;
+                         method_index++) {
+                        if (ngx_strncasecmp(
+                              ngx_http_lua_ordered_methods[method_index]->data,
+                              lua_method_name,
+                              method_name_length)
+                            == 0) {
+                            break;
+                        }
+                    }
 
-            } else {
-                if (type != LUA_TNUMBER) {
+                    if (method_index == methods_number) {
+                        /* unknown method */
+                        method = NGX_HTTP_UNKNOWN;
+                        ngx_method_name = ngx_palloc(r->pool,
+                                                     sizeof(ngx_str_t));
+                        if (ngx_method_name == NULL) {
+                            return luaL_error(L, "out of memory");
+                        }
+                        ngx_method_name->data = ngx_palloc(r->pool,
+                                                           method_name_length);
+                        if (ngx_method_name->data == NULL) {
+                            return luaL_error(L, "out of memory");
+                        }
+                        ngx_memcpy(ngx_method_name->data,
+                                   lua_method_name,
+                                   method_name_length);
+                        ngx_memcpy(ngx_method_name->data + method_name_length,
```

Comment:
I don't think it's wise to use `ngx_memcpy` for a single character assignment here. Better use the array index form:

```
ngx_method_name[method_name_length] = ' ';
```

What do you think?